flink-core/src/main/java/org/apache/flink/api/common/operators/base/ReduceOperatorBase.java

```diff
@@ -202,7 +203,9 @@ public void setCustomPartitioner(Partitioner<?> customPartitioner) {
 
 		if (inputColumns.length > 0) {
 			boolean[] inputOrderings = new boolean[inputColumns.length];
-			TypeComparator<T> inputComparator = ((CompositeType<T>) inputType).createComparator(inputColumns, inputOrderings, 0, executionConfig);
+			TypeComparator<T> inputComparator = inputType instanceof BasicTypeInfo
+					? ((BasicTypeInfo<T>) inputType).createComparator(inputOrderings[0], executionConfig)
```

Comment:
Cast to `AtomicType`. `inputOrderings[0]` can be replaced by `false`, since the array fields are not explicitly initialized.